research/delf/delf/python/training/README.md

```diff
@@ -118,11 +118,116 @@ files can take up to 12 hours and up to 500 GB of space disk.*
+  --export_path=gldv2_model
 ```
+
+## Testing the Trained Model
+
+After the trained model has been exported, it can be used to extract DELF features from 2 images 
+of the same landmark and to perform a matching test between the 2 images based on the extracted 
+features to validate they represent the same landmark.
+
+Start by downloading the Oxford buildings dataset:
+````
+mkdir data && cd data
+wget http://www.robots.ox.ac.uk/~vgg/data/oxbuildings/oxbuild_images.tgz
+mkdir oxford5k_images oxford5k_features
+tar -xvzf oxbuild_images.tgz -C oxford5k_images/
+cd ../
+echo data/oxford5k_images/hertford_000056.jpg >> list_images.txt
+echo data/oxford5k_images/oxford_000317.jpg >> list_images.txt
+````
+
+Make a copy of the [`delf_config_example.pbtxt`](../examples/delf_config_example.pbtxt) 
+protobuffer file which configures the DELF feature extraction. Update the file by making the
+following changes:
+* set the `model_path` attribute to the directory containing the exported model, `gldv2_model`
+  in this example
+* add at the root level the attribute `is_tf2_exported` with the value `true`
+* set to `false` the `use_pca` attribute inside `delf_local_config`
+
+The ensuing file should resemble the following:
+```
+model_path: "gldv2_model"
+image_scales: .25
+image_scales: .3536
+image_scales: .5
+image_scales: .7071
+image_scales: 1.0
+image_scales: 1.4142
+image_scales: 2.0
+is_tf2_exported: true
+delf_local_config {
+  use_pca: false
+  # Note that for the exported model provided as an example, layer_name and
+  # iou_threshold are hard-coded in the checkpoint. So, the layer_name and
+  # iou_threshold variables here have no effect on the provided
+  # extract_features.py script.
+  layer_name: "resnet_v1_50/block3"
+  iou_threshold: 1.0
+  max_feature_num: 1000
+  score_threshold: 100.0
+  pca_parameters {
+    mean_path: "parameters/delf_gld_20190411/pca/mean.datum"
+    projection_matrix_path: "parameters/delf_gld_20190411/pca/pca_proj_mat.datum"
+    pca_dim: 40
+    use_whitening: false
+  }
+}
+```
+
+Run the following command to extract DELF features for the images `hertford_000056.jpg` and
+`oxford_000317.jpg`:
+```
+python ../examples/extract_features.py \
+  --config_path delf_config_example.pbtxt \
+  --list_images_path list_images.txt \
+  --output_dir data/oxford5k_features
+```
+
+Run the following command to perform feature matching between the images `hertford_000056.jpg` 
+and `oxford_000317.jpg`:
+```
+python ../examples/match_images.py \
+  --image_1_path data/oxford5k_images/hertford_000056.jpg \
+  --image_2_path data/oxford5k_images/oxford_000317.jpg \
+  --features_1_path data/oxford5k_features/hertford_000056.delf \
+  --features_2_path data/oxford5k_features/oxford_000317.delf \
+  --output_image matched_images.png
+```
+
+The generated image `matched_images.png` should look similar to this one:
+![MatchedImagesExample](../examples/matched_images_example.jpg)
```

Comment:
just curious, when you did these steps, was the image also similar to this one? (I see you're reusing the visualization from the existing one)

Reply:
The matching image was extremely similar, yes, so the model trains correctly. That's why I linked to the already existing image:) But just for completeness I uploaded my test image and linked to it. You can see the 2 results are very similar.